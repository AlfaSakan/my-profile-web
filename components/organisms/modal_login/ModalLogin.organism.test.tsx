import renderer from 'react-test-renderer';
import ModalLogin from './ModalLogin.organism';

describe('HeaderBody component test', () => {
  it('should render correctly', () => {
    const onChangePhoneNumber = jest.fn();
    const onChangeName = jest.fn();
    const onClickLogin = jest.fn();

    const tree = renderer
      .create(
        <ModalLogin
          name="name"
          onChangePhoneNumber={onChangePhoneNumber}
          isOpen
          onChangeName={onChangeName}
          phoneNumber="8123456789"
          onClickLogin={onClickLogin}
          textErrorName="textErrorName"
          textErrorPhone="textErrorPhone"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
