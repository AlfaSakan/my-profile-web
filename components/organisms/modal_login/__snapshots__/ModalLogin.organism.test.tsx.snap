// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBody component test should render correctly 1`] = `
<div
  className="fixed h-full w-full bg-black-200 top-0 flex items-center justify-center"
>
  <div
    className="md:w-1/2 md:mx-0 h-1/2 w-full mx-10 rounded-2xl flex flex-col overflow-hidden bg-white"
    onClick={[Function]}
  >
    <div
      className="flex items-center justify-between px-8 py-4 bg-secondary"
    >
      <p
        className="text-sm"
      >
        Login / Register
      </p>
    </div>
    <div
      className="flex flex-1 flex-col px-8 py-4 border-y gap-8"
    >
      <div>
        <p
          className="text-sm"
        >
          Nama
        </p>
        <input
          className="w-full focus:outline-none rounded-md py-2 px-4 border py-3 mt-2"
          onChange={[MockFunction]}
          placeholder="Type a message"
          type="text"
          value="name"
        />
        <div
          className="flex items-center absolute mt-1"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "fill": "rgb(239 68 68)",
                "fontSize": 14,
                "marginRight": 4,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
            />
          </svg>
          <p
            className="text-red-500 text-xs"
          >
            textErrorName
          </p>
        </div>
      </div>
      <div>
        <p
          className="text-sm"
        >
          Nomor Handphone
        </p>
        <input
          className="w-full focus:outline-none rounded-md py-2 px-4 border py-3 mt-2"
          onChange={[MockFunction]}
          placeholder="Type a message"
          type="text"
          value="8123456789"
        />
        <div
          className="flex items-center absolute mt-1"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "fill": "rgb(239 68 68)",
                "fontSize": 14,
                "marginRight": 4,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
            />
          </svg>
          <p
            className="text-red-500 text-xs"
          >
            textErrorPhone
          </p>
        </div>
      </div>
    </div>
    <div
      className="flex items-center justify-end px-8 py-4"
    >
      <button
        className="flex items-center justify-center w-32 h-12 rounded-lg bg-secondary"
        onClick={[MockFunction]}
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      >
        <p
          className="text-sm"
        >
          Login
        </p>
      </button>
    </div>
  </div>
</div>
`;
